main/src/ca/uwaterloo/flix/Main.scala

```diff
@@ -224,12 +225,23 @@ object Main {
           Console.println(results.output(flix.getFormatter))
         }
       case Validation.Failure(errors) =>
-        errors.sortBy(_.source.name).foreach(e => println(e.message(flix.getFormatter)))
+        mkMessage(errors.sortBy(_.source.name), flix.getFormatter)
+          .foreach(println)
         println()
         println(s"Compilation failed with ${errors.length} error(s).")
         System.exit(1)
     }
 
+    /**
+      * Converts a list of compiler error messages to a list of printable messages.
+      * Decides whether or not to print the explanation.
+      */
+    def mkMessage(errors: LazyList[CompilationMessage], formatter: Formatter): List[String] = {
```

Comment:
This is the same `mkMessage`. What about putting it somewhere? Maybe in the Flix object?